test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -132,30 +141,43 @@ def test_framework_version_cpu(cpu):
 
 # TODO: Enable as canary once resource cleaning lambda is added
 @pytest.mark.model("N/A")
-@pytest.mark.parametrize("ec2_instance_type", ['p2.xlarge'], indirect=True)
-def test_framework_version_gpu(gpu, ec2_connection):
+@pytest.mark.parametrize("ec2_instance_type", ["p2.xlarge"], indirect=True)
+def test_framework_and_cuda_version_gpu(gpu, ec2_connection):
     """
-    Check that the framework version in the image tag is the same as the one on a running container.
+    Check that the framework  and cuda version in the image tag is the same as the one on a running container.
 
     :param gpu: ECR image URI with "gpu" in the name
     :param ec2_connection: fixture to establish connection with an ec2 instance
     """
     image = gpu
-    if "tensorflow-inference" in image:
-        pytest.skip(msg="TF inference does not have core tensorflow installed")
-
     tested_framework, tag_framework_version = get_framework_and_version_from_tag(image)
 
-    # Module name is "torch"
-    if tested_framework == "pytorch":
-        tested_framework = "torch"
-    cmd = f'import {tested_framework}; print({tested_framework}.__version__)'
-    output = ec2.execute_ec2_training_test(ec2_connection, image, cmd, executable="python")
-
-    if is_canary_context():
-        assert tag_framework_version in output.stdout.strip()
+    # Framework Version Check #
+    # Skip framework version test for tensorflow-inference, since it doesn't have core TF installed
+    if "tensorflow-inference" not in image:
+        # Module name is "torch"
+        if tested_framework == "pytorch":
+            tested_framework = "torch"
+        cmd = f"import {tested_framework}; print({tested_framework}.__version__)"
+        output = ec2.execute_ec2_training_test(ec2_connection, image, cmd, executable="python")
+
+        if is_canary_context():
+            assert tag_framework_version in output.stdout.strip()
+        else:
+            assert tag_framework_version == output.stdout.strip()
+
+    # CUDA Version Check #
+    cuda_version = re.search(r"-cu(\d+)-", image).group(1)
+
+    # MXNet inference containers do not currently have nvcc in $PATH, so check cuda directories
+    if "mxnet-inference" in image:
+        cuda_cmd = "bash -c 'ls /usr/local/cuda*'"
```

Comment:
I purposefully left it this way because this would require interpolating the period inside the cuda version, which previously you were not comfortable with. Do you want me to revert the behavior?

Reply:
Changed behavior in new revision to use readlink